MailCore/Cache/MailboxManager/MailboxManager+Draft.swift

```diff
@@ -158,4 +185,32 @@ public extension MailboxManager {
             }
         }
     }
+
+    func moveScheduleToDraft(draftResource: String) async {
+        await moveScheduleToDraft(draftAction: draftResource.appending("/schedule"))
+    }
+
+    func moveScheduleToDraft(draftAction: String) async {
+        do {
+            try await apiFetcher.deleteSchedule(draftAction: draftAction)
+            if let scheduledDraftsFolder = getFolder(with: .scheduledDrafts) {
+                await refreshFolderContent(scheduledDraftsFolder.freezeIfNeeded())
+            }
+        } catch {
+            print(error)
+        }
+    }
+
+    func loadRemotely(from draftResource: String) async -> Draft? {
+        do {
+            let draft = try await apiFetcher.draft(draftResource: draftResource)
+            try? writeTransaction { realm in
+                realm.add(draft, update: .modified)
+            }
+            return draft.freeze()
+        } catch {
+            print(error)
```

Comment:
This should rethrow. And maybe we should do something similar as with drafts ? -> Delete locally